clang/lib/CodeGen/CGHLSLRuntime.h

```diff
@@ -96,9 +136,15 @@ class CGHLSLRuntime {
                                    BufferResBinding &Binding);
   void addConstant(VarDecl *D, Buffer &CB);
   void addBufferDecls(const DeclContext *DC, Buffer &CB);
+  llvm::Triple::ArchType getArch();
   llvm::SmallVector<Buffer> Buffers;
 };
 
+// Default implementation
+template <unsigned BI> llvm::Intrinsic::ID CGHLSLRuntime::getIntrinsic() {
+  static_assert(false, "getIntrinsic is only allowed on specialized templates");
```

Comment:
So if we don't define a default then we leave it up to the linker to error which is a less transparent error message:

```
ld.lld: error: undefined symbol: unsigned int clang::CodeGen::CGHLSLRuntime::getIntrinsic<627u>()
>>> referenced by CGBuiltin.cpp:18199 (/mnt/DevDrive/projects/llvm-project/clang/lib/CodeGen/CGBuiltin.cpp:18199)
>>>               CGBuiltin.cpp.o:(clang::CodeGen::CodeGenFunction::EmitHLSLBuiltinExpr(unsigned int, clang::CallExpr const*)) in archive lib/libclangCodeGen.a
>>> did you mean: unsigned int clang::CodeGen::CGHLSLRuntime::getIntrinsic<626u>()
>>> defined in: lib/libclangCodeGen.a(CGBuiltin.cpp.o)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

By adding the default it instead looks like this which makes it more clear that we have to specialize `BI__builtin_hlsl_elementwise_any`  if we want to use `getIntrinsic`.

```
/mnt/DevDrive/projects/llvm-project/clang/lib/CodeGen/CGHLSLRuntime.h:146:19: error: static assertion failed: getIntrinsic is only allowed on specialized templates
  146 |     static_assert(false, "getIntrinsic is only allowed on specialized templates");
      |                   ^~~~~
/mnt/DevDrive/projects/llvm-project/clang/lib/CodeGen/CGBuiltin.cpp:18199:14: note: in instantiation of function template specialization 'clang::CodeGen::CGHLSLRuntime::getIntrinsic<627U>' requested here
 18199 |             .getIntrinsic<Builtin::BI__builtin_hlsl_elementwise_any>(), ArrayRef<Value *>{Op0}, nullptr, "dx.any");
       |              ^
1 error generated.
```